Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -379,7 +379,9 @@ class TrackStateProxy {
   /// @param srf Shared pointer to the surface to set
   /// @note This overload is only present in case @c ReadOnly is false.
   template <bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
-  void setReferenceSurface(std::shared_ptr<const Surface> srf) {
+  void setReferenceSurface(
+      std::shared_ptr<const Surface>
+          srf) {  // NOLINT(performance-unnecessary-value-param)
```

Comment:
not sure why this is necessary... clang believes the `std::move(srf)` below does not do anything

Reply:
I think we have a implicit move here, see https://stackoverflow.com/questions/9779079/why-does-c11-have-implicit-moves-for-value-parameters-but-not-for-rvalue-para
So clang-tidy might be correct and we can remove the `std::move`...